python/tank_vendor/shotgun_base/shotgun_path.py

```diff
@@ -0,0 +1,232 @@
+
+    @classmethod
+    def from_system_dict(cls, system_dict):
+        """
+        Creates a path from data contained in a dictionary keyed by
+        sys.platform constants.
+
+        :param system_dict: Dictionary with possible keys
+                        win32, darwin and linux2.
+        :return: ShotgunPath instance
+        """
+        windows_path = system_dict.get("win32")
+        linux_path = system_dict.get("linux2")
+        macosx_path = system_dict.get("darwin")
+
+        return cls(windows_path, linux_path, macosx_path)
+
+    @classmethod
+    def from_current_os_path(cls, path):
+        """
+        Creates a path object for a path on the current platform only.
+
+        :param path: Path on the current os platform.
+        :return: ShotgunPath instance
+        """
+        windows_path = None
+        linux_path = None
+        macosx_path = None
+
+        if sys.platform == "win32":
+            windows_path = path
+        elif sys.platform == "linux2":
+            linux_path = path
+        elif sys.platform == "darwin":
+            macosx_path = path
+        else:
+            return ValueError("Unsupported platform '%s'." % sys.platform)
+
+        return cls(windows_path, linux_path, macosx_path)
+
+    def __init__(self, windows_path=None, linux_path=None, macosx_path=None):
+        """
+        Constructor
+
+        :param windows_path: Path on windows to associate with this path object
+        :param linux_path: Path on linux to associate with this path object
+        :param macosx_path: Path on mac to associate with this path object
+        """
+        self._windows_path = sanitize_path(windows_path, "\\")
+        self._linux_path = sanitize_path(linux_path, "/")
+        self._macosx_path = sanitize_path(macosx_path, "/")
+
+    def __repr__(self):
+        return "<Path win:'%s', linux:'%s', mac:'%s'>" % (
+            self._windows_path,
+            self._linux_path,
+            self._macosx_path
+        )
+
+    def __eq__(self, other):
+        """
+        Test if this ShotgunPath instance is equal to the other ShotgunPath instance
+
+        :param other:   The other ShotgunPath instance to compare with
+        :returns:       True if path is same is other, false otherwise
+        """
+
+        if not isinstance(other, ShotgunPath):
+            return NotImplemented
+
+        return self.mac == other.mac and self.windows == other.windows and self.linux == other.linux
+
+    def __ne__(self, other):
+        """
+        Test if this path is not equal to the given path
+
+        :param other:   Other ShotgunPath instance to compare with
+        :returns:       True if self != other, False otherwise
+        """
+        is_equal = self.__eq__(other)
+        if is_equal is NotImplemented:
+            return NotImplemented
+        return not is_equal
+
+    @property
+    def mac(self):
+        """
+        The macosx representation of the path
```

Comment:
Technically, it's not called Mac anymore but OSX. Apple dropped the Mac part of the branding a while back. :p